src/hooks/useHttpClient.tsx

```diff
@@ -62,6 +64,7 @@ export const HttpClientContextProvider = ({
     responseInterceptorId = axiosInstance.interceptors.response.use(
       undefined,
       async function (error: Error) {
+        checkConnection?.();
```

Comment:
This seems a little concerning to me but it is probably okay for a first pass.

I view Toasts (and other forms of pop up messaging) as a design choice that sdk users should be able to decide on. By adding checkConnection to the http client, we are locking in sdk users in to having toasts in their app. The context provider is good because it seems like an sdk user could override the toast so it doesn't display, or make it look native to their app, but it doesn't give them a way to handle the scenario differently (splash page, logout, modal, status icon, etc.)

I feel like this might be a great place for optional dependencies that @markdlv mentioned at one point which would allow us to use toasts as the default behavior, if installed, or leave the handling up to the sdk user entirely.

Reply:
👍 Thanks for those thoughts, excellent points. I will look into this.